lib/semian/simple_enum.rb

```diff
@@ -0,0 +1,29 @@
+module Semian
+  module Simple
+    class Enum #:nodoc:
+      def initialize(symbol_list:)
+        @allowed_symbols = symbol_list
+        reset
+      end
+
+      attr_reader :value
+
+      def value=(sym)
+        raise ArgumentError unless @allowed_symbols.include?(sym)
+        @value = sym
+      end
+
+      def increment(val = 1)
+        @value = @allowed_symbols[(@allowed_symbols.index(@value) + val) % @allowed_symbols.size]
+      end
```

Comment:
I might need `SysV::Enum` to inherit from `SysV::Integer` if byroot's suggestion goes through https://github.com/Shopify/semian/pull/62#discussion-diff-44469194 because thats the only way it can be done if `#value=` disappears or goes private